dpnp/tests/test_mathematical.py

```diff
@@ -1143,6 +1143,152 @@ def test_complex(self, xp):
         assert_raises((ValueError, TypeError), xp.i0, a)
 
 
+class TestInterp:
+    @pytest.mark.parametrize(
+        "dtype_x", get_all_dtypes(no_bool=True, no_complex=True)
+    )
+    @pytest.mark.parametrize("dtype_y", get_all_dtypes(no_bool=True))
+    def test_all_dtypes(self, dtype_x, dtype_y):
+        x = numpy.linspace(0.1, 9.9, 20).astype(dtype_x)
```

Comment:
Do we have any test where `x` and `xp` have different dtypes?

Reply:
done